shared/fs/destination-picker/container.js

```diff
@@ -57,16 +67,21 @@ const canWrite = memoize(
 
 const canCopy = memoize(
   (stateProps, ownProps: OwnProps) =>
-    canWrite(stateProps, ownProps) &&
-    getDestinationParentPath(stateProps, ownProps) !==
-      Types.getPathParent(stateProps._moveOrCopy.sourceItemPath)
+    canWrite(stateProps, ownProps) && (
+      stateProps._destinationPicker.source.type === 'incoming-share' || (
+        stateProps._destinationPicker.source.type === 'move-or-copy' &&
+        // $FlowIssue can't figure out that we're in a MoveOrCopySource here, for whatever reason.
```

Comment:
I think this is because `getDestinationParentPath` returns either a `Type.Path` or a `string`. There's no way flow can figure out which one it is.

Why not just do `Types.getPathParent` and `Types.getLocalPathParent` respectively instead of `getDestinationParentPath`?

Reply:
It actually always returns a path.

Reply:
We don't ever want the local path here. This is only for the destination, which we only supply as the source by default in a `move-or-copy` dialog. In any other situation we just want it to be `/keybase`.

Reply:
You're right; sorry I mis-read.